language-server/modules/langserver-commons/src/main/java/org/ballerinalang/langserver/commons/LanguageServerLightWeightMode.java

```diff
@@ -0,0 +1,10 @@
+package org.ballerinalang.langserver.commons;
+
+/**
+ * Interface to represent the Language Server is in LightWeightMode.
+ *
+ * @since 2201.4.0
+ */
+public class LanguageServerLightWeightMode {
```

Comment:
We don't need a new class like this. The `InitializeParams` already has a parameter called `enableLightWeightMode`. See `org.ballerinalang.langserver.BallerinaLanguageServer#isLightWeightMode()` method for an example usage. However, we have a better approach.

At `BallerinaLanguageServer:88`, we create an instance of `LSClientCapabilitiesImpl` containing an instance of `InitializationOptions`. We should add the light weight mode boolean as a new method to that interface. And in the other places, use that to determine if we are in lightweight mode.